src/components/Input.tsx

```diff
@@ -0,0 +1,24 @@
+import { useState} from 'react';
+import Button from './Button';
+
+
+const Input=()=> {
+    const [value, setValue] = useState<string>("");
+
+    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+        setValue(event.currentTarget.value);
+      }
+    function handleClick()
+    {
+      !!value && console.log(value);
+      setValue("");
+    }
```

Comment:
A couple points here!

1) I forgot to mention this earlier, but you might need to make sure that value exists before setting a state.
2) Please use arrow functions for callbacks.

```suggestion
  const handleClick = () => 
  {
if (value) {
      console.log(value);
      setValue("");
    }
    }
```

Reply:
Understood, have made the changes